File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
     int n = nums.Length;
     Dictionary<int,int> map = new Dictionary<int,int>();
        int rsum = 0;
        map.Add(0,1);
        int count = 0;
     for(int i = 0;i<n;i++){
            rsum = rsum + nums[i];
            int diff = rsum - k;

            if(map.ContainsKey(diff)){
                count = count + map[diff];  
            }

            if(!map.ContainsKey(rsum)){
                map.Add(rsum , 1);
            }else{
                map[rsum]++;
            }
     }

    
     return count;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        

        unordered_map<int,int> map;
        int count = 0;
        map[0] = 1;
        int rsum = 0;
        for(int i = 0;i<nums.size();i++){
            rsum = rsum +nums[i];
            int diff = rsum - k;
            if(map.find(diff)!=map.end()){
                count = count + map[diff];
            }

            if(map.find(rsum)!= map.end()){
                map[rsum]++;
            }else{
                map[rsum] = 1;
            }
        }

        return count;
    }
};